File: app/src/main/java/com.example.safehome/Door/BiFoldDoorFactory.java
package com.example.safehome.Door;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


public class BiFoldDoorFactory implements DoorFactory<Door> {
    private static final Logger log = LoggerFactory.getLogger(BiFoldDoorFactory.class);

    /**
     * FactoryMethod returns class of type Door.
     *
     * @param request
     * @return FactoryMethod returns class of type Door.
     */
    @Override
    public Door factoryMethod(Door request) {
        return null;
    }
}

File: app/src/main/kotlin/com.example.safehome/Door/BiFoldDoorFactory.kt
package com.example.safehome.Door

import org.slf4j.Logger
import org.slf4j.LoggerFactory

class BiFoldDoorFactory : DoorFactory<Door?> {
    /**
     * FactoryMethod returns class of type Door.
     *
     * @param request
     * @return FactoryMethod returns class of type Door.
     */
    override fun factoryMethod(request: Door?): Door? {
        return null
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(BiFoldDoorFactory::class.java)
    }
}